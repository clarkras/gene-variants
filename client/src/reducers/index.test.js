/*global jest, it, expect*/

import rootReducer from './index';

it('has the correct initial state', () => {
  const state = rootReducer(undefined, {type: undefined });

  // noinspection JSUnresolvedFunction
  expect(state).toMatchSnapshot();
});

it('handles SELECT_GENE', () => {
  const selectedGene = 'GENE_1';
  const state = rootReducer({ foo: 123 }, {type: 'SELECT_GENE', selectedGene });

  // noinspection JSUnresolvedFunction
  expect(state).toMatchSnapshot();
});

it('handles SET_VARIANTS', () => {
  const variants = [
    'variant_1',
    'variant_2',
  ];
  const state = rootReducer({ foo: 123 }, {type: 'SET_VARIANTS', variants });

  // noinspection JSUnresolvedFunction
  expect(state).toMatchSnapshot();
});

// etc.
