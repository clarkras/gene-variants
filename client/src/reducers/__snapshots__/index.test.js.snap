// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles SELECT_GENE 1`] = `
Object {
  "foo": 123,
  "selectedGene": undefined,
}
`;

exports[`handles SET_VARIANTS 1`] = `
Object {
  "foo": 123,
  "variants": Array [
    "variant_1",
    "variant_2",
  ],
}
`;

exports[`has the correct initial state 1`] = `
Object {
  "genes": Array [],
  "groupBy": "Alias",
  "selectedGene": null,
  "variants": Array [],
}
`;
